services/server/src/server/session.ts

```diff
@@ -0,0 +1,115 @@
+import expressSession from "express-session";
+import createMemoryStore from "memorystore";
+import { Pool } from "pg";
+import genFunc from "connect-pg-simple";
+import config from "config";
+import session from "express-session";
+
+import logger from "../common/logger";
+import { NextFunction, Request, Response } from "express";
+
+function initMemoryStore() {
+  const MemoryStore = createMemoryStore(expressSession);
+
+  logger.info("Using memory based session");
+  return new MemoryStore({
+    checkPeriod: config.get("session.maxAge"),
+  });
+}
+
+function initDatabaseStore() {
+  const pool = new Pool({
+    host: process.env.SOURCIFY_POSTGRES_HOST,
+    database: process.env.SOURCIFY_POSTGRES_DB,
+    user: process.env.SOURCIFY_POSTGRES_USER,
+    password: process.env.SOURCIFY_POSTGRES_PASSWORD,
+    port: parseInt(process.env.SOURCIFY_POSTGRES_PORT || "5432"),
+  });
+
+  // This listener is necessary otherwise the sourcify process crashes if the database is closed
+  pool.prependListener("error", () => {
+    logger.error("Database connection lost for session pool");
+  });
+
+  const PostgresqlStore = genFunc(expressSession);
+
+  logger.info("Using database based session");
+  return new PostgresqlStore({
+    pool: pool,
+    // Pruning expired sessions every 15 minutes
+    pruneSessionInterval: 900,
+  });
+}
+
+function getSessionStore() {
+  const sessionStoreType = config.get("session.storeType");
+
+  switch (sessionStoreType) {
+    case "database": {
+      if (
+        process.env.SOURCIFY_POSTGRES_HOST &&
+        process.env.SOURCIFY_POSTGRES_DB &&
+        process.env.SOURCIFY_POSTGRES_USER &&
+        process.env.SOURCIFY_POSTGRES_PASSWORD
+      ) {
+        return initDatabaseStore();
+      } else {
+        // Log the error but continue with the memory session
+        // This is needed because tests all use the same local-test.js file but not all the tests have the database enabled
+        logger.error(
```

Comment:
Throw here

Reply:
I think this is necessary because not all tests run with the database enabled but the `local-test.js` file contains `memory: "database"`. I don't know how to deal with this, ideas?

Reply:
My point is, if the user chose "database" and they haven't given the necessary values, it must throw. Souricy can't run with missing info.

For the tests, don't all they have the env vars set?